src/lightning/fabric/wrappers.py

```diff
@@ -385,3 +388,26 @@ def _capture(model: Any, **kwargs: Any) -> Any:
 
 if _TORCH_GREATER_EQUAL_2_0:
     torch.compile = _capture_compile_kwargs(torch.compile)
+
+
+def _register_backward_hook(tensor: Tensor) -> Tensor:
+    from lightning.fabric.fabric import _in_fabric_backward
+
+    # strategy_requires = is_overridden("backward", self._strategy, parent=Strategy)
+    # precision_requires = any(
+    #     is_overridden(method, self._precision, parent=Precision)
+    #     for method in ("pre_backward", "backward", "post_backward")
+    # )
+    # TODO: How do we get a reference to strategy and precision here?
+    strategy_requires = True
+    precision_requires = True
```

Comment:
Probably have to pass strategy to the FabricModule?

Reply:
In https://github.com/Lightning-AI/pytorch-lightning/pull/18844/files#diff-7cf8c282d7fd9f34b838a3267dcd70a4d7e2434480b656144b631c3cb3620313R106 I had to do it. But then you have to make it a weak reference in the hook in case the tensor is still used when fabric is no longer around.

Reply:
Yes, that's probably unavoidable.